File: Extra/leetcode/14. Longest Common Prefix/solution.c
//hell nah
#include <stdio.h>
#include <stdlib.h>

char * longestCommonPrefix(char ** strs, int strsSize){

  int commonsize = 2;
  char* common = malloc(sizeof(char) * commonsize);
  int j = 0;
  char curr;

  for (int i = 0; i < strsSize; i++, j++) {
    if (i == commonsize) {
      commonsize *= 2;
      realloc(common, commonsize);
    }

    if (i == 0) curr = *(*(strs) + j);

    if (*(*(strs + i) + j) == '\0') {
      break;
    } else if (*(*(strs + i) +j ) != *(*(strs + i+1) + j)) {
      break;
    } else {
      
    }

  }

  return common;
}

File: Extra/leetcode/14. Longest Common Prefix/Solution.java
class Solution {
  public String longestCommonPrefix(String[] strs) {
    StringBuilder common = new StringBuilder();

    String first = strs[0];

    for (int i = 0; i < first.length(); i++) {
      char c = first.charAt(i);

      for (int j = 1; j < strs.length; j++) {
        //System.out.println(strs[j].length() + " " + i + " " + strs[j].charAt(i) + " " + c);
        if (strs[j].length() <= i || strs[j].charAt(i) != c) return common.toString();
      }

      common.append(first.charAt(i));
    }
    return common.toString();
  }
}

